### PR TITLE
fix: show no modal message when collector submits offer on unlisted price artworks

### PR DESCRIPTION
**Pull request by ashleyjelks**
This fixes an issue where users were warned when their offer exceeded the listed (but hidden) price. Users could keep lowering the offer until the modal no longer appears, effectively guessing the listed price when the gallery intended for it to not be displayed. Now, when a price is unlisted we simply show nothing when users make an offer.

https://artsyproduct.atlassian.net/browse/PURCHASE-2965

### CODE REVIEW

**Review by araujobarret — APPROVED** [review]
Looks good!

File: src/v2/Apps/Order/Routes/Offer/index.tsx [hunk 1]
```diff
@@ -168,6 +168,9 @@ export class OfferRoute extends Component<OfferProps, OfferState> {
     }
 
     const listPriceCents = this.props.order.totalListPriceCents
+    const artworkPrice = this?.props?.order?.lineItems?.edges?.[0]?.node
+      ?.artwork?.price
+    const isPriceHidden = isNil(artworkPrice) || artworkPrice === ""
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/Offer/index.tsx:173]: I'm not sure if you need `isNil` here 🤔 
In theory, if the price is `null`, `undefined`, or `""` then just doing `!artworkPrice` should return the expected value, no?

File: src/v2/Apps/Order/Routes/Respond/index.tsx [hunk 2]
```diff
@@ -117,6 +118,9 @@ export class RespondRoute extends Component<RespondProps, RespondState> {
     } = this.state
 
     const search = this.props.match?.location.search || ""
+    const artworkPrice = this?.props?.order?.lineItems?.edges?.[0]?.node
+      ?.artwork?.price
+    const isPriceHidden = isNil(artworkPrice) || artworkPrice === ""
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/Respond/index.tsx:123]: Same as the other comment relating to `isNil`

File: src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx [hunk 3]
```diff
@@ -142,6 +148,28 @@ describe("Offer InitialMutation", () => {
...
+
+    it("shows no modal warning when an offer made on work with hidden price", async () => {
+      await page.setOfferAmount(2000000)
+      await page.clickSubmit()
+      await page.expectButtonSpinnerWhenSubmitting()
+      await page.expectNoModal()
+      expect(mutations.mockFetch).toHaveBeenCalled()
+    })
+  })
+
```
Comment (jo-rs) [inline src/v2/Apps/Order/Routes/__tests__/Offer.jest.tsx:172]: Lovely 🥇

**Review by jo-rs — APPROVED** [review]
Looks great! 💯 
Just a small question in regards to the usage of `isNil`

**Comment by dzucconi** [comment]
For whatever reason this PR seems to have broken the build?

**Comment by dzucconi** [comment]
Can confirm this is failing locally with:

```
src/v2/Apps/__tests__/Fixtures/Order.ts:594:14 - error TS2552: Cannot find name 'OfferArtworkOrEditionSetkNode_Artwork'. Did you mean 'OfferArtworkOrEditionSetNode_Artwork'?

594           ...OfferArtworkOrEditionSetkNode_Artwork,
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/v2/Apps/__tests__/Fixtures/Order.ts:166:7
    166 const OfferArtworkOrEditionSetNode_Artwork = {
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    'OfferArtworkOrEditionSetNode_Artwork' is declared here.

src/v2/Apps/__tests__/Fixtures/Order.ts:596:14 - error TS2304: Cannot find name 'OrderArtworkShipmentNode'.

596           ...OrderArtworkShipmentNode,
                 ~~~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors.
```

-----

We shouldn't be writing fixtures at all any more. Any new tests should just be using the Relay auto-mocks — or is there a particular reason for the fixtures here?

**Pull request merged**